core/types/governance_new_market.go

```diff
@@ -330,6 +330,10 @@ func NewMarketConfigurationFromProto(p *vegapb.NewMarketConfiguration) (*NewMark
 }
 
 func SuccessorConfigFromProto(p *vegapb.SuccessorConfiguration) *SuccessorConfig {
+	// successor config is optional, but make sure that, if provided, it's not set to empty parent market ID
+	if len(p.ParentMarketId) == 0 {
```

Comment:
How would a checkpoint file get into this state? In the protos the successor config is a new optional field, so for a checkpoint file taken from mainnet pre-all this code it would unmarshal into a nil-pointer.

And in new code any market proposal with a successor config and no parent ID would fail validation and never exist to be in a checkpoint.